src/Chip/Icon.js

```diff
@@ -13,9 +13,16 @@ export const Icon = ({ icon }) => {
                     width: 24px;
                     height: 24px;
                     margin-left: 4px;
+                    margin-right: -6px;
                     border-radius: 50%;
                     overflow: hidden;
                 }
+
+                /* force icons to fit in our icon drawer */
+                span > :global(*) {
```

Comment:
If possible change this to a more specific selector, as css it's interpreted from right to left, so now is very expensive. (as @Mohammer5 pointed out to me a few weeks ago)

Reply:
I don't know how we can predict what will go in here?

Reply:
I'd be happy to change this if there are any alternatives. We can "guess" that most likely there will be an `<img>` or `<svg>` tag passed in, but it's hard to guarantee it. We could just ignore it and let the sub element overflow?

It doesn't look too nice though.

On the performance side. I'm not advocating that we should make it slow, speed is a priority. But CSS selector speed is much, _much_, **much** faster than triggering layout reflows which [a lot of CSS properties do](https://csstriggers.com/). A short list of the expensive ones which trigger layout thrashing:

```
    border-radius: 5px;
    padding: 3px;
    box-shadow: 0 5px 5px #000;
    transform: rotate(10deg);
```
 
I do not think we need to worry too much about CSS selector speed as long as the selectors are kept reasonable:

E.g. I think `span > *` is OK while something like `div:hover[data-foo="bar"].name:nth-child(5)::after` would **not** be OK.

Some articles backing up my claims:

- https://benfrain.com/css-performance-revisited-selectors-bloat-expensive-styles
- https://calendar.perfplanet.com/2011/css-selector-performance-has-changed-for-the-better/
- http://perfectionkills.com/profiling-css-for-fun-and-profit-optimization-notes/
- http://www.stevesouders.com/blog/2009/03/10/performance-impact-of-css-selectors/

Note that these range from 2009-2014 so a lot of those benchmarks are done on _very_ slow browsers like IE6-9 and Firefox 3.0. I've re-run the benchmarks I could easily run and selector speed today is a lot faster than it was.

Layout reflows are still expensive.

Reply:
Thinking on this some more, it's not certain that the aspect ratio will be 1:1 for all elements passed in as `icon`, so I think we had it good enough with just the overflow. Added a docstring instead.